modules/dag-l1/src/main/scala/org/tessellation/dag/l1/http/p2p/clients/L0GlobalSnapshotClient.scala

```diff
@@ -0,0 +1,36 @@
+package org.tessellation.dag.l1.http.p2p.clients
+
+import cats.effect.Async
+
+import org.tessellation.dag.snapshot.{GlobalSnapshot, SnapshotOrdinal}
+import org.tessellation.ext.codecs.BinaryCodec.decoder
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.sdk.http.p2p.PeerResponse
+import org.tessellation.sdk.http.p2p.PeerResponse.PeerResponse
+import org.tessellation.security.SecurityProvider
+import org.tessellation.security.signature.Signed
+
+import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
+import org.http4s.client.Client
+
+trait L0GlobalSnapshotClient[F[_]] {
+  def getLatestOrdinal: PeerResponse[F, SnapshotOrdinal]
+  def get(ordinal: SnapshotOrdinal): PeerResponse[F, Signed[GlobalSnapshot]]
+}
+
+object L0GlobalSnapshotClient {
+
+  def make[F[_]: Async: SecurityProvider: KryoSerializer](client: Client[F]): L0GlobalSnapshotClient[F] =
+    new L0GlobalSnapshotClient[F] {
+
+      def getLatestOrdinal: PeerResponse[F, SnapshotOrdinal] =
+        PeerResponse[F, SnapshotOrdinal]("global-snapshot/latest/ordinal")(client)(
+          circeEntityDecoder,
```

Comment:
here I'd do the same. I'd avoid passing implicits explicitly as much as possible